src/main/java/seedu/task/model/ModelManager.java

```diff
@@ -255,7 +255,7 @@ public String toString() {
         public boolean run(ReadOnlyTask task) {
             return taskKeyWords.stream()
                     .filter(keyword -> StringUtil.containsIgnoreCase(task.getTask().fullName, keyword) 
-                    		|| StringUtil.containsIgnoreCase(task.getDescription().value, keyword))
+                    		|| StringUtil.containsIgnoreCase(task.getDescription().orElse(null).toString() , keyword))
```

Comment:
task.getDescription() returns Optional<Description>. I tried orElse("") but because it is not Optional<String>, the empty string is not accepted. Also, a null.toString() actually returns empty string. Otherwise, if we dont use null, we need to create a new Description with empty string? something like: task.getDescription().orElse(new Description("")).toString()  ??